src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,68 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final int dimensionLimit = 100;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figures figure = getRandomFigureFromEnum();

```

Comment:
You can simplify your logic and avoid creating the Figures enum by using random figure index
private int figureIndex = random.nextInt(FIGURE_TYPE_COUNT);
where FIGURE_TYPE_COUNT is count of your figures and use switch - case also